protostar-cli/src/lib.rs

```diff
@@ -0,0 +1,80 @@
+use anyhow::{anyhow, Result};
+use camino::Utf8PathBuf;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use starknet::{
+    accounts::SingleOwnerAccount,
+    core::{chain_id, types::FieldElement},
+    providers::jsonrpc::{HttpTransport, JsonRpcClient},
+    signers::{LocalWallet, SigningKey},
+};
+use std::fs;
+use url::Url;
+
+#[derive(Deserialize, Serialize)]
+struct Account {
+    private_key: String,
+    public_key: String,
+    address: String,
+}
+
+pub enum Network {
+    Testnet,
+    Testnet2,
+    Mainnet,
+}
+
+impl Network {
+    pub fn get_chain_id(&self) -> FieldElement {
+        match self {
+            Network::Testnet => chain_id::TESTNET,
+            Network::Testnet2 => chain_id::TESTNET2,
+            Network::Mainnet => chain_id::MAINNET,
+        }
+    }
+    pub fn get_value(&self) -> &'static str {
+        match self {
+            Network::Testnet => "alpha-goerli",
+            Network::Testnet2 => "alpha-goerli2",
+            Network::Mainnet => "alpha-mainnet,",
+        }
+    }
+}
+
+pub fn get_provider(url: &str) -> Result<JsonRpcClient<HttpTransport>> {
+    let parsed_url = Url::parse(url)?;
+    let provider = JsonRpcClient::new(HttpTransport::new(parsed_url));
+    Ok(provider)
+}
+
+fn get_account_info(name: &str, chain_id: &str, path: &Utf8PathBuf) -> Result<Account> {
+    let accounts: Value = serde_json::from_str(&fs::read_to_string(path)?)?;
```

Comment:
Why are we doing deserialization to `Value` first instead of defining a struct for that and using it directly?

Reply:
Right, changed it to use Account struct